Add a Warning to Avoid Julia's Broadcatisng in Expressions (#717)
jump-dev/Convex.jl: docs/src/manual/types.md

# Once the problem is solved, we can call evaluate() on expr:
evaluate(expr)
```

> !!! warning Julia's broadcasting, using the `.` operator should be avoided. The expression `-log(1.0 .+ A * x)` will fail while `-log(1.0 + A * x)` will be correctly encoded. 

Reviewer: I think we need to format it like this:
```suggestion

!!! warning
    Julia's broadcasting, using the `.` operator should be avoided. The expression `-log(1.0 .+ A * x)` will fail while `-log(1.0 + A * x)` will be correctly encoded. 
```
xref https://documenter.juliadocs.org/stable/showcase/#Admonitions

Author: I actually was wondering about this. Decided just to follow the current warning in https://github.com/jump-dev/Convex.jl/blob/master/docs/src/manual/advanced.md.
Yet now I see the rendered MarkDown failed me :-).

evaluate(expr)
```

!!! warning

>     Julia's broadcasting, using the `.` operator should be avoided. The expression `-log(1.0 .+ A * x)` will fail while `-log(1.0 + A * x)` will be correctly encoded. 

Reviewer 2: ```suggestion
    Avoid using Julia's broadcasting (the `.` operator). As an example, the
    expression `-log(1.0 .+ A * x)` will fail while `-log(1.0 + A * x)` will
    succeed.
```
